src/jobs/send-out-monthy-income-report.ts

```diff
@@ -29,27 +29,92 @@ const sendOutMonthlyIncomeReport = async () => {
       orderBy: { datePurchased: "asc" },
     });
 
-    const mappedArtists = keyBy(allArtists, "id");
+    const artistOwners = new Map<number, number>();
+    const usersById = new Map<
+      number,
+      {
+        id: number;
+        name: string | null;
+        email: string | null;
+      }
+    >();
+
+    for (const artist of allArtists) {
+      if (typeof artist.id === "number" && typeof artist.userId === "number") {
+        artistOwners.set(artist.id, artist.userId);
+      }
+
+      if (typeof artist.userId === "number" && artist.user) {
+        usersById.set(artist.userId, artist.user);
+      }
+    }
+
+    const groupedSales = groupBy(sales, (sale) => {
+      const userIdFromSale = sale.artist?.userId;
+      if (typeof userIdFromSale === "number") {
+        return String(userIdFromSale);
+      }
+
+      const artistId = sale.artist?.id ?? (sale as { artistId?: number }).artistId;
+      if (typeof artistId === "number") {
+        const ownerId = artistOwners.get(artistId);
+        if (typeof ownerId === "number") {
+          return String(ownerId);
+        }
+      }
 
-    const groupedSales = groupBy(sales, "artist.userId");
-    for (const [userId, userSales] of Object.entries(groupedSales)) {
-      if (userSales.length === 0) {
+      return "unknown";
+    });
+
+    for (const [userIdKey, userSales] of Object.entries(groupedSales)) {
+      if (!userSales.length) {
+        continue;
+      }
+
+      const userId = Number(userIdKey);
+
+      if (!Number.isFinite(userId)) {
+        logger.warn(
+          `Skipping monthly income report for unresolved owner identifier ${userIdKey}`
+        );
+        continue;
+      }
+
+      const user = usersById.get(userId);
+
+      if (!user?.email) {
+        logger.warn(
+          `Skipping monthly income report for artist owner ${userId} because no email address was found.`
+        );
+        continue;
+      }
+
+      const normalizedSales = userSales.map((sale) => ({
```

Comment:
I don't think amount can be anything other than a number because it comes from the database. So this feels unnecessary to me.